app/main.py

```diff
@@ -4,31 +1,65 @@
+from typing import List, Tuple, Dict
+
 
+class Deck:
+    def __init__(self, row: int, column: int,
+                 is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: Tuple[int, int], end: Tuple[int, int],
+                 is_drowned: bool = False) -> None:
+        self.decks: List[Deck] = []
+        self.is_drowned = is_drowned
+        self.create_decks(start, end)
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def create_decks(self, start: Tuple[int, int],
+                     end: Tuple[int, int]) -> None:
+        start_row, start_col = start
+        end_row, end_col = end
+        if start_row == end_row:
+            for col in range(start_col, end_col + 1):
+                self.decks.append(Deck(start_row, col))
+        elif start_col == end_col:
+            for row in range(start_row, end_row + 1):
+                self.decks.append(Deck(row, start_col))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
 
+    def fire(self, row: int, column: int) -> bool:
+        deck = self.get_deck(row, column)
+        if deck and deck.is_alive:
+            deck.is_alive = False
```

Comment:
This condition checks if the deck is alive before marking it as not alive. This is correct, but the subsequent check for all decks being not alive should be inside the same condition block to avoid unnecessary checks when the deck is already not alive.